core/src/main/java/com/datastax/oss/driver/internal/core/util/concurrent/CompletableFutures.java

```diff
@@ -100,7 +100,10 @@ public static <T> CompletionStage<Void> allSuccessful(List<CompletionStage<T>> i
                 } else {
                   Throwable finalError = errors.get(0);
                   for (int i = 1; i < errors.size(); i++) {
-                    finalError.addSuppressed(errors.get(i));
+                    Throwable suppressedError = errors.get(i);
+                    if (finalError != suppressedError) {
+                      finalError.addSuppressed(suppressedError);
+                    }
```

Comment:
Took me a minute to convince myself that != is correct over equals() here.  I couldn't find any docs on whether self-suppression was tied to reference equality or object equality but the more reference equality does seem like the more conservative approach here.